ksql-common/src/main/java/io/confluent/ksql/name/Name.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.name;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.util.Identifiers;
+import java.util.Objects;
+
+/**
+ * The base type for all names, which just wraps a String in
+ * a type-safe wrapper and supplies formatting options.
+ *
+ * @param <T> ensure type safety of methods
+ */
+@Immutable
+public abstract class Name<T extends Name<?>> {
+
+  protected final String name;
+
+  protected Name(final String name) {
+    this.name = Identifiers.ensureTrimmed(Objects.requireNonNull(name, "name"), "name");
+  }
+
+  // we should remove this getter after all code has
+  // migrated to use Name instead of Strings to make
+  // sure that we never lose type safety
+  public String name() {
+    return name;
+  }
+
+  /**
+   * @see String#equalsIgnoreCase(String)
+   */
+  public boolean equalsIgnoreCase(final T o) {
```

Comment:
I take it the plan is to remove this method as well in time?

Reply:
hehe - I am just about to post a PR that does this!